components/org.wso2.carbon.config/src/test/java/org/wso2/carbon/config/UtilsTest.java

```diff
@@ -112,4 +112,19 @@ public void testPathSubstitution(String path, String pathSeparator) {
         Assert.assertEquals(ConfigurationUtils.substituteVariables(config),
                 path + pathSeparator + "deployment" + pathSeparator);
     }
+
+    @Test
+    public void testEscapeSpecialCharacters() {
+        String inputString = "C:\\Software\\WSO2*Kernal-5.0.0\\wso2\\worker\\bin";
```

Comment:
We can use a data provider to test different scenarios of input strings. In that case, we can merge the 2 test cases. WDYT?